syntax = "proto3";

message Message {
    string cmd = 1;
    int64 target_messenger_id = 2;
    int64 sender_messenger_id = 3;
    string job_name = 4;
    map<string, string> cmd_info = 5;
    repeated uint64 fids = 6;
    repeated MsgVector vecs = 7;
    uint64 timestamp = 8;
    uint64 reply_timestamp = 9;
    int64 target_node_id = 10;
    int64 sender_node_id = 11;
}

message MsgVector {
    int32 size = 1;
    repeated double values = 2;
}
